src/js/stores/UserSettingsStore.ts

```diff
@@ -2,7 +2,8 @@ import EventEmitter from "events";
 
 import LocalStorageUtil from "../utils/LocalStorageUtil";
 
-const LOCAL_STORAGE_KEY = "dcosUserSettings";
+const LOCAL_STORAGE_KEY: string = "dcosUserSettings";
+const JSON_EDITOR_SETTING: string = "JSONEditor";
```

Comment:
we have other stuff that stores data in the `UserSettingsStore`. that other stuff knows the key to store stuff by and does not introduce an extra method within `UserSettingsStore`. do we have a reason to introduce a new pattern here?

Reply:
Since I have converted the store to typescript I was considering using the existing functions with a return type of `unknown` but I considered that this would result into an awful lot of unnecessary code ensuring that something is of a certain type. And frankly speaking I was trying to set a good example by creating an abstraction with a clear return value and handling the type validation within those functions. Also I did not need to pass the store key to the components who would need the setting. 

So yes I had a reason to introduce the new pattern. That reason was a better abstraction and type validation which otherwise would have been unnecessary code duplication within this PR.

Reply:
now that makes sense. thanks for explaining. we could create a follow-up issue to convert the "old style" to the new as that does not look like a lot of work.